graphix/_linalg.py

```diff
@@ -0,0 +1,307 @@
+            If `True`, the RREF matrix is copied into a new instance, otherwise `self` is modified. Defaults to `False`.
+
+        Returns
+        -------
+        mat_ref: MatGF2
+            The matrix in row-reduced echelon form.
+        """
+        ncols_value = self.shape[1] if ncols is None else ncols
+        mat_ref = self.copy() if copy else self
+
+        return MatGF2(_elimination_jit(mat_ref, ncols=ncols_value, full_reduce=True))
+
+
+def solve_f2_linear_system(mat: MatGF2, b: MatGF2) -> MatGF2:
+    r"""Solve the linear system (LS) `mat @ x == b`.
+
+    Parameters
+    ----------
+    mat : MatGF2
+        Matrix with shape `(m, n)` containing the LS coefficients in row echelon form (REF).
+    b : MatGF2
+        Matrix with shape `(m,)` containing the constants column vector.
+
+    Returns
+    -------
+    x : MatGF2
+        Matrix with shape `(n,)` containing the solutions of the LS.
+
+    Notes
+    -----
+    This function is not integrated in `:class: graphix.linalg.MatGF2` because it does not perform any checks on the form of `mat` to ensure that it is in REF or that the system is solvable.
+    """
+    return MatGF2(_solve_f2_linear_system_jit(mat, b))
+
+
+@nb.njit("uint8[::1](uint8[:,::1], uint8[::1])")
+def _solve_f2_linear_system_jit(
+    mat_data: npt.NDArray[np.uint8], b_data: npt.NDArray[np.uint8]
+) -> npt.NDArray[np.uint8]:
+    """See docstring of `:func:solve_f2_linear_system` for details."""
+    m, n = mat_data.shape
+    x = np.zeros(n, dtype=np.uint8)
+
+    # Find first row that is all-zero
+    for i in range(m):
+        for j in range(n):
+            if mat_data[i, j] == 1:
+                break  # Row is not zero → go to next row
+        else:
+            m_nonzero = i  # No break: this row is all-zero
+            break
+    else:
+        m_nonzero = m
+
+    # Backward substitution from row m_nonzero - 1 to 0
+    for i in range(m_nonzero - 1, -1, -1):
+        # Find first non-zero column in row i
+        pivot = -1
+        for j in range(n):
+            if mat_data[i, j] == 1:
+                pivot = j
+                break
+
+        # Sum x_k for k such that mat_data[i, k] == 1
+        acc = 0
+        for k in range(pivot, n):
+            if mat_data[i, k] == 1:
+                acc ^= x[k]
+
+        x[pivot] = b_data[i] ^ acc
+
+    return x
+
+
+@nb.njit("uint8[:,::1](uint8[:,::1], uint64, boolean)")
+def _elimination_jit(mat_data: npt.NDArray[np.uint8], ncols: int, full_reduce: bool) -> npt.NDArray[np.uint8]:
+    r"""Return row echelon form (REF) or row-reduced echelon form (RREF) by performing Gaussian elimination.
+
+    Parameters
+    ----------
+    mat_data : npt.NDArray[np.uint8]
+        Matrix to be gaussian-eliminated.
+    n_cols : int
+        Number of columns over which to perform Gaussian elimination.
+    full_reduce : bool
+        Flag determining the operation mode. Output is in RREF (respectively, REF) if `True` (repectively, `False`).
```

Comment:
The second occurrence of `respectively` misses an `s`.

I would say:
 ```suggestion
        Flag determining the operation mode. Output is in RREF if `True`, REF otherwise.
```

Reply:
Done in [56fd0a0](https://github.com/TeamGraphix/graphix/pull/337/commits/56fd0a0fdd832782ebcd6573130df939e7a8e79a)